tools/debug-ui/src/NetworkInfoView.tsx

```diff
@@ -148,8 +195,183 @@ export const NetworkInfoView = ({ addr }: NetworkInfoViewProps) => {
+            </tr>
+        </table>
+
+        <table className="peers">
+            <thead>
+                <tr>
+                    <th>Address</th>
+                    <th>Validator?</th>
+                    <th>Account ID</th>
+                    <th>Last ping</th>
+                    <th>Height</th>
+                    <th>Last Block Hash</th>
+                    <th>Tracked Shards</th>
+                    <th>Archival</th>
+                    <th>Connection type</th>
+                    <th>First connection</th>
+                    <th>Traffic (last minute)</th>
+                    <th>Route to validators</th>
+                </tr>
+            </thead>
+            <tbody>
+                {peers.map(({ peer, validator, routedValidator, statusClassName }) => {
+                    return <tr key={peer.peer_id}>
+                        <td>{addDebugPortLink(peer.addr)}</td>
+                        <td>{validator.join(', ')}</td>
+                        <td>{peer.peer_id.substring(8, 14)}...</td> {/* strips prefix 'ed25519:' */}
+                        <td className={
+                            peer.last_time_received_message_millis > 60 * 1000 ? 'peer_far_behind' : ''
+                        }>{formatDurationInMillis(peer.last_time_received_message_millis)}</td>
+                        <td className={statusClassName}>{peer.height}</td>
+                        <td>{peer.block_hash}</td>
+                        <td>{JSON.stringify(peer.tracked_shards)}</td>
+                        <td>{JSON.stringify(peer.archival)}</td>
+                        <td>{peer.is_outbound_peer ? 'OUT' : 'IN'}</td>
+                        <td>{formatDurationInMillis(peer.connection_established_time_millis)}</td>
+                        <td>{formatTraffic(peer.received_bytes_per_sec, peer.sent_bytes_per_sec)}</td>
+                        <td><CollapsableValidatorList validators={routedValidator} /></td>
+                    </tr>
+                })}
+            </tbody>
+        </table>
+        {peerStorageView}
     </div>;
-};
\ No newline at end of file
+};
+
+function toHumanTime(seconds: number): string {
+    let result = "";
+    if (seconds >= 60) {
+        let minutes = Math.floor(seconds / 60);
+        seconds = seconds % 60;
+        if (minutes > 60) {
+            let hours = Math.floor(minutes / 60);
+            minutes = minutes % 60;
+            if (hours > 24) {
+                let days = Math.floor(hours / 24);
+                hours = hours % 24;
+                result += days + " days ";
+            }
+            result += hours + " h ";
+        }
+        result += minutes + " m ";
+    }
+    result += seconds + " s"
+    return result;
+}
+
+function PeerStorageView(addr: string): ReactElement {
+    const [showPeerStorage, setShowPeerStorage] = useState(false);
+
+    const { data: peerStore, error, isLoading } =
+        useQuery(['peer_store', addr], () => fetchPeerStore(addr));
```

Comment:
BTW - this data is quite large (that's why I've put it behind a button on the debug page).

Is useQuery lazy? or does it start fetching immediately?